src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTokenExtensions.cs

```diff
@@ -168,6 +170,14 @@ public static bool IsBeginningOfStatementContext(this SyntaxToken token)
 
                 case SyntaxKind.ElseKeyword:
                     return true;
+
+            }
+
+            if (token.ValueText == "await")
```

Comment:
This should likely be `Text` since `@await` is not an await token (whereas this check would match). But why not just a simple `IsKind` check?

Reply:
The kind is "Identifier" in this situation.

Reply:
I believe that's because you're keyword recommender tests are not inside an async method.

Reply:
Indeed. Thanks!